<section>
	<h1>UCSF ChimeraX</h1>
	<ul>
		<li>Target audience</li>
		<ul>
			<li>Scientists who need to analyze and model molecular <br/>systems in some detail</li>
			<li>Scientists who want to develop new analytical methods</li>
			<li>Scientists who need to combine experimental modalities</li>
			<li>Scientists or educators who wish to communicate <br/>biological concepts visually</li>
		</ul>
		<li>Organization</li>
		<ul>
			<li>Core features</li>
			<li>Additional tools</li>
		</ul>
	</ul>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>ChimeraX Core Features</h1>
	<ul>
		<li>Concepts</li>
		<ul>
			<li>Graphical manipulation</li>
			<li>Chemical knowledge</li>
			<ul>
				<li>Hierarchy: Model&rarr;Chain&rarr;Residue&rarr;Atom</li>
				<li>Attributes available at model, residue, and atom levels</li>
			</ul>
			<li>Selection - Action</li>
			<li>GUI/Command-line equivalence</li>
			<ul>
				<li>For core, not necessarily extensions</li>
			</ul>
		</ul>
	</ul>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<ul>
		<li><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/ui.html#mousemode" target="_blank">Using the mouse</a></li>
		<ul>
			<li>Pointer over atom</li>
			<ul><li>“Tooltip” with atom and residue information</li></ul>
			<li>Left-button</li>
			<ul>
				<li>X-Y Rotation (on virtual sphere)</li>
				<li>Z Rotation (outside of virtual sphere)</li>
			</ul>
			<li>Middle-button</li>
			<ul><li>X-Y Translation</li></ul>
			<li>Right-button (configurable using <b>Right Mouse</b> toolbar</li>
			<li>Control+</li>
			<ul>
				<li>Left-button: select (click or drag)</li>
			</ul>
		</ul>
	</ul>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<ul style="font-size: 90%">
		<li><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/selection.html" target="_blank">Selection methods</a></li>
		<ul>
			<li>Using mouse</li>
			<ul>
				<li>Control-click and control-drag</li>
				<li>Shift-control-click and shift-control-drag to add to selection</li>
			</ul>
			<li>By atom and residue names and types</li>
			<li>By higher order structure</li>
			<ul><li>e.g., chains, ligands, solvent, protein, nucleic acid</li></ul>
			<li>By spatial location</li>
			<ul><li>e.g., within some distance of another selection</li></ul>
			<li>By attribute values</li>
			<ul><li>e.g., temperature factor, occupancy</li></ul>
			<li>Can extend any selection with up-arrow: &uarr;</li>
			<li>NOTE: by default, if nothing is selected, everything is selected</li>
		</ul>
	</ul>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%">
	<p><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/tools/toolbar.html" target="_blank">Toolbars</a> 
	provide quick mechanism to perform various commands.  In general, they operate on the selected structures as appropriate.  
	There are currently 6 toolbars:</p>
	<ul>
		<li><b>Home</b>: this toolbar contains a set of the most common actions.  In the future, users will be able to add
			their favorite action to this toolbar</li>
		<li class="fragment" data-fragment-index="2"><b>Molecule Display</b>: provides a set of actions to change the display of a molecular structure and perform some structural analysis 
			(e.g. show hydrogen bonds).</li>
		<li class="fragment" data-fragment-index="3"><b>Nucleotides</b>: supports changing the display of nucleotide elements within a structure.</li>
		<li class="fragment" data-fragment-index="4"><b>Graphics</b>: changes various graphics values such as the background, lighting type, camera modes, etc.</li>
		<li class="fragment" data-fragment-index="5"><b>Map</b>: controls for manipulating map (volume) data</li>
		<li class="fragment" data-fragment-index="6"><b>Right Mouse</b>: settings for operation of the right mouse. Allows the right mouse to be used for a large number of purposes
		including movement of structures, manipulation of maps and structures, and annotations.</li>
	</ul>
	<div class="fragment fade-out" data-fragment-index="2" style="position: absolute; x:50px;y:100px">
		<img  height="60px" src="modules/chimerax-intro/HomeToolbar.png"/>
	</div>
	<div class="fragment fade-out" data-fragment-index="3" style="position: absolute; x:50px;y:100px">
		<img class="fragment fade-in" height="60px" data-fragment-index="2" src="modules/chimerax-intro/MoleculeToolbar.png"/>
	</div>
	<div class="fragment fade-out" data-fragment-index="4" style="position: absolute; x:50px;y:100px">
		<img class="fragment fade-in" height="60px" data-fragment-index="3" src="modules/chimerax-intro/NucleotideToolbar.png"/>
	</div>
	<div class="fragment fade-out" data-fragment-index="5" style="position: absolute; x:50px;y:100px">
		<img class="fragment fade-in" height="60px" data-fragment-index="4" src="modules/chimerax-intro/GraphicsToolbar.png"/>
	</div>
	<div class="fragment fade-out" data-fragment-index="6" style="position: absolute; x:50px;y:100px">
		<img class="fragment fade-in" height="60px" data-fragment-index="5" src="modules/chimerax-intro/MapToolbar.png"/>
	</div>
	<div style="position: absolute; x:50px;y:100px">
		<img class="fragment fade-in" height="60px" data-fragment-index="6" src="modules/chimerax-intro/RightMouseToolbar.png"/>
	</div>
	</div>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:65%">
		<p>Currently, most actions are performed using 
		<a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/index.html#commands" target="_blank">commands</a> entered
		on the ChimeraX <a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/tools/cli.html" target="_blank">
			Command Line Interface</a>.  To get a list of the available commands, enter <b>usage</b> into the command line.
		To get more information about a command enter <b>help</b> <i>command</i> and ChimeraX's web browser will be displayed
		with the documentation for that command. Some of the most commonly used commands are described 
		in the next slides, but the first it's important to know how ChimeraX refers to models, chains, residues, and atoms.</p>
	</div>
</section>

<section style="text-align:left">
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:65%">
	<h4 style="text-align:left"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/atomspec.html" target="_blank">
			Target specification</a></h4>
				<table style="font-size:70%">
					<tr>
						<th>Symbol</th><th>Reference Level</th><th>Definition</th><th>Examples</th>
					</tr>
					<tr>
						<td>#</td>
						<td>model</td>
						<td>model number assigned to the data in ChimeraX<br/>
							(hierarchical, with positive integers separated by dots: N, N.N, N.N.N, etc.) </td>
						<td><b>#1<br/>#1.3</b></td>
					</tr>
					<tr>
						<td>/</td>
						<td>chain</td>
						<td>
							chain identifier (case-insensitive unless both upper- and lowercase chain IDs are present)
						</td>
						<td><b>/A</b></td>
					</tr>
					<tr>
						<td>:</td>
						<td>residue</td>
						<td>
							residue number<br/><i>OR</i><br/>
residue name (case-insensitive)
						</td>
						<td><b>:51<br/>:glu</b></td>
					</tr>
					<tr>
						<td>@</td>
						<td>atom</td>
						<td>
atom name(case-insensitive)
						</td>
						<td><b>@ca</b></td>
					<tr>
				</table>
				<div style="font-size:70%">
					<p><b>Lists and Ranges</b><br/>
				Multiple model or residues numbers can be entered as comma-separated lists and/or one or more ranges.  For example:
					<ul style="margin-left: 30px; margin-top: -15px">
						<li><b>#1/B-D,F</b>: chains B,C,D, and F in model 1</li>
						<li><b>:start-40</b>: residue numbers up to 40 in all chains</li>
						<li><b>:12-25,48@ca,n</b>: atoms named CA and N in residues 12-25 and 48
				</ul>
				<p>There are many more features in the target specification for ChimeraX, including wildcarding, 
				built-in classifications such as <b>solvent, ions, ligand, protein, nucleic, sidechain, helix, strand</b> etc.  One very
				commonly use built-in is <b>sel</b> which refers to the current selection (e.g. <b>color sel blue</b>).<p>
				</div>
				
	</div>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
		<h4><highlight>Basic commands</highlight></h4>
			<dl style="font-size: 80%">
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/delete.html" target="_blank">delete</a>:</dt>
				<dd>deletes the specified atoms, bonds, or pseudobonds.  Note: this is not undoable!</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/open.html" target="_blank">open</a>,
				<a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/close.html">close</a>,
				<a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/save.html">save</a>:</dt>
				<dd>
				The <b>open</b> command reads data in several input formats (e.g PDB/mmCIF, DICOM, Gromacs, MRC, and many others) from local files, urls, and online databases.  This command is also used to read ChimeraX session files (.cxs).
				The <b>close</b> command removes models from ChimeraX, including any submodels.
				The <b>save</b> command saves the current session or map.  It is also used to save images from the current display.
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/select.html">select</a>:</dt>
				<dd>
				 A selection specifies items (models or their parts) for subsequent operations; the selected items 
				 are highlighted with green outlines in the graphics window. The select command can:
				 <ul>
					 <li>make a new selection of specified items</li>
					 <li>modify an existing selection</li>
					 <li>select by zone (distance cutoff from specified items)</li>
				 </ul>
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/show.html">show,hide</a> (disp,~disp):</dt>
				<dd>
				The commands <b>show</b> and <b>hide</b> display and undisplay the specified atoms, bonds, pseudobonds, cartoons, molecular 
				surfaces, and entire models. Following the usual convention, a blank specification means “all.”  Both commands take an 
				optional <b>target</b> argument that limits the effect to atoms (<b>a</b>), bonds (<b>b</b>), pseudobonds (<b>p</b>), cartoons (<b>c</b>), surfaces(<b>s</b>) or
				entire models (<b>m</b>).
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/undo.html">undo,redo</a>:</dt>
				<dd>
				The <b>undo</b> and <b>redo</b> commands provide a limited undo/redo capability.  At this point, only <b>show, hide, color, 
					size, style, cartoon,</b> and <b>select</b> are supported.  In some cases there are restrictions on even those commands.  
				We expect this to be extended in the future.
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/view.html">view</a>:</dt>
				<dd>
				The <b>view</b> commands changes the window focus to the specied items or "all".
				</dd>
			</dl>
	</div>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
		<h4><highlight>Depiction commands</highlight></h4>
			<dl style="font-size: 80%">
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/cartoon.html" target="_blank">cartoon</a> (ribbon):</dt>
				<dd>
				The cartoon or ribbon command shows a simplified backbone representation of the specified biopolymer residues.  The <b>cartoon style</b>
				subcommand provides several options for the style of the representation, including width, thickness, and shape.
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/nucleotides.html" target="_blank">nucleotides</a></dt>
				<dd>
				The nucleotides command generates special representations of nucleic acids, with choices for mode: <b>atoms, fill, ladder, stubs, tube/slab, slab</b>
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/style.html" target="_blank">style</a>:</dt>
				<dd>
				The style command sets atom/bond display styles and the number of dashes per pseudobond. The style can be one of the following: <b>sphere, stick, </b>or <b>ball</b>.  The style command also provides options for ring fill as well as pseudobond dashes.
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/surface.html" target="_blank">surface</a>:</dt>
				<dd>
				The surface command creates and displays molecular surfaces, either atomically detailed solvent-excluded surfaces (default) or lower-resolution Gaussian surfaces.  Surface display style (solid, mesh, or dot) can be adjusted with <b>surface style</b>.  There are several options that support
				how the surface is created, including the atoms enclosed by the surface and the atoms incuded in the surface.
				</dd>
				<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/color.html" target="_blank">color</a>:</dt>
				<dd>
				Depending on the type of coloring, color may apply to atomic models and their parts, including cartoon and molecular 
				surface representations, and volume data (map) displays.  Subcommands include <b>color sequential</b> (rainbow) to support rainbow
				coloring of atomic models, <b>color modify</b> to adjust the color for atomic models, <b>color byattribute</b> to color 
				based on an atom, residue, or model attribute, and coloring by map values: <b>color sample, color electrostatic, </b>
				and <b>color gradient</b>.  There are also subcommands to color surfaces by distance (<b>color radial</b>, <b>color cylindrical</b>,
				<b>color height</b> and to color by zone (<b>color zone</b>).
				</dd>
			</dl>
	</div>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
		<h4><highlight>Analysis commands</highlight></h4>
		<dl>
			<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/hbonds.html" target="_blank">hbonds</a>:</dt>
			<dd>
			The hbonds command uses atom types and geometric criteria to identify hydrogen bonds (H-bonds) involving the specified atoms. 
			</dd>
			<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/matchmaker.html" target="_blank">matchmaker</a> (mmaker):</dt>
			<dd>
			The matchmaker (or mmaker) command superimposes
			protein or nucleic acid structures by first
			creating a pairwise sequence alignment,
			then fitting the aligned residue pairs.
			</dd>
			<dt style="float:left; margin-right: 5px"><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/morph.html" target="_blank">morph</a>:</dt>
			<dd>
			 The morph command creates a trajectory that morphs between two or more atomic models. 
			 The trajectory can be played back with the 
			 <a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/coordset.html" target="_blank">coordset</a>
			 command or the slider graphical interface.
			</dd>
		</dl>
	</div>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
	<h4 style="text-align:left"><highlight>Useful tools</highlight></h4>
		<ul>
			<li><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/tools/log.html" target="_blank">Log</a></li>
			<ul>
				<li>Echoes commands that have been executed, with any truncations expanded</li>
				<li>Shows messages and other output information, potentially including hyperlinks and
					command-execution links</li>
				<li>Context menu and 
					<a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/commands/log.html" target="_blank">log</a> command
					provide support for:
					<ul>
						<li>Saving the log to an HTML file</li>
						<li>Adding a thumbmnail image to the log</li>
						<li>Adding metadata associated with a model to the log</li>
						<li>Setting whether echoed commands should link to documentation or be execution links</li>
					</ul>
				</li>
			</ul>
	<div style="position: absolute; top:100px; left: 100px; width: 80%" data-autoslide="2000" class="fragment fade-in">
		<img class="plain" src="modules/chimerax-intro/log1.png">
	</div>
	<div style="position: absolute; top:100px; left: 100px; width: 80%" class="fragment fade-in">
		<img class="plain" src="modules/chimerax-intro/log2.png">
	</div>
	</div>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
	<h4 style="text-align:left"><highlight>Useful tools</highlight></h4>
		<ul>
			<li style="Color: lightblue">Log</li>
			<li><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/tools/modelpanel.html" target="_blank">Model Panel</a></li>
			<ul>
				<li>Shows model hierarchy of currently open models</li>
				<li>Allows selective display of models</li>
				<li>Can show/hide models or submodels</li>
			</ul>
		</ul>
	<div style="position: absolute; top:100px; left: 100px; width: 80%" data-autoslide="2000" class="fragment fade-in">
		<img class="plain" src="modules/chimerax-intro/modelPanel.png">
	</div>
	</div>
	<aside class="notes">
	</aside>
</section>


<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
	<h4 style="text-align:left"><highlight>Useful tools</highlight></h4>
		<ul>
			<li style="Color: lightblue">Log</li>
			<li style="Color: lightblue">Model Panel</li>
			<li><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/tools/sideview.html" target="_blank">Side View</a></li>
			<ul>
				<li>Helps locate model in Z direction</li>
				<li>Provides access to hither-and-yon clipping planes</li>
			</ul>
		</ul>
	</ul>
	<div style="position: absolute; top:100px; left: 100px; width: 80%" data-autoslide="2000" class="fragment fade-in">
		<img class="plain" src="modules/chimerax-intro/sv1.png">
	</div>
	<div style="position: absolute; top:100px; left: 100px; width: 80%" class="fragment fade-in">
		<img class="plain" src="modules/chimerax-intro/sv2.png">
	</div>
	<aside class="notes">
	</aside>
</section>


<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
	<h4 style="text-align:left"><highlight>Useful tools</highlight></h4>
		<ul>
			<li style="Color: lightblue">Log</li>
			<li style="Color: lightblue">Model Panel</li>
			<li style="color: lightblue">Side View</li>
			<li><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/tools/sequenceviewer.html" target="_blank">Sequence Viewer</a></li>
			<ul>
				<li>Displays sequence of currently loaded structures</li>
				<li>Selection linked to structure</li>
				<li>Can display alignments</li>
				<li>Alignment headers can show conservation (AL2CO) as well as consensus</li>
			</ul>
		</ul>
	<div style="position: absolute; top:100px; left: 100px; width: 80%" class="fragment fade-in">
		<img class="plain" src="modules/chimerax-intro/sequence.png">
	</div>
	</div>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Core Features (cont.)</h1>
	<div style="height:850px;font-size:70%; text-align:left">
	<h4 style="text-align:left"><highlight>Useful tools</highlight></h4>
		<ul>
			<li style="Color: lightblue">Log</li>
			<li style="Color: lightblue">Model Panel</li>
			<li style="color: lightblue">Side View</li>
			<li style="color: lightblue">Sequence</li>
			<li><a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/tools/basicactions.html" target="_blank">Basic actions</a></li>
			<ul>
				<li>Quick access to:
					<ul>
						<li>select named sets of atoms</li>
						<li>show/hide or color atoms</li>
						<li>show/hide or color cartoon segments</li>
						<li>show/hide or color molecular surface patches</li>
					</ul>
				</li>
			</ul>
		</ul>
	<div style="position: absolute; top:100px; left: 100px; width: 80%" class="fragment fade-in">
		<img class="plain" src="modules/chimerax-intro/basicactions.png">
	</div>
	</div>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Toolshed</h1>
	<div style="height:850px;font-size:70%; text-align:left">
		<p>The <a href="https://cxtoolshed.rbvi.ucsf.edu/apps/" target="_blank">ChimeraX toolshed</a> provides a mechanism
		for other developers to add new tools and commands to ChimeraX.  Several additional tools are already available on the
		toolshed, so users should check there for new content.  Also, the UCSF team will release updates to individual ChimeraX
		components by publishing them to the toolshed.</p>
	<div style="position: absolute; top:130px; left: 50px" class="fragment fade-in">
		<img src="modules/chimerax-intro/toolshed.png">
	</div>
	<aside class="notes">
	</aside>
</section>


<section>
	<h1>Scripting</h1>
	<div style="height:850px;font-size:70%; text-align:left">
	<ul>
		<li>Scripting is executing a set of commands from a file</li>
		<ul>
			<li>Useful for repetitive tasks</li>
			<li>Can serve as documentation</li>
		</ul>
		<li>ChimeraX commands may be read in from a file or typed directly in the command line interface</li>
		<li>The ChimeraX browser combines scripting with simple HTML to create web interactions with ChimeraX.  To use,
			simply add command links (e.g. "&lt;a href="cxcmd:open 1hgg"&gt;open 1hgg&lt;/a&gt;") in your html and open
			the file using ChimeraX's web browser.
	</ul>
	<div style="position: absolute; top:130px; left: 50px" class="fragment fade-in">
		<img src="modules/chimerax-intro/html1.png">
	</div>
	<div style="position: absolute; top:130px; left: 50px" class="fragment fade-in">
		<img src="modules/chimerax-intro/html2.png">
	</div>
	</div>
	<aside class="notes">
	</aside>
</section>

<section>
	<h1>Getting Help</h1>
	<div style="height:850px;font-size:70%; text-align:left">
		<ul>
			<li>The best, most up-to-date source for help is the user's guide.  It is available directly from
				the ChimeraX help menu (<b>Help&rarr;User Guide</b>) or on the web
				<a href="https://www.rbvi.ucsf.edu/chimerax/docs/user/index.html" target="_blank">https://www.rbvi.ucsf.edu/chimerax/docs/user/</a>.
			</li>
			<li>You can also subscribe to the e-mail list: 
				<a href="http://plato.cgl.ucsf.edu/mailman/listinfo/chimerax-users" target="_blank">ChimeraX-users</a>, which is regularly
				monitored by the ChimeraX team.
			</li>
		</ul>
	</div>
</section>

